strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/repository/MavenGroupRepositoryComponent.java

```diff
@@ -0,0 +1,166 @@
+package org.carlspring.strongbox.repository;
+
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.providers.io.RepositoryPath;
+import org.carlspring.strongbox.providers.layout.LayoutProvider;
+import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
+import org.carlspring.strongbox.services.ArtifactMetadataService;
+import org.carlspring.strongbox.services.ConfigurationManagementService;
+import org.carlspring.strongbox.storage.metadata.MavenMetadataManager;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.mutable.MutableBoolean;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author Przemyslaw Fusik
+ */
+@Component
+public class MavenGroupRepositoryComponent
+{
+
+    @Inject
+    private ConfigurationManagementService configurationManagementService;
+
+    @Inject
+    private ConfigurationManager configurationManager;
+
+    @Inject
+    private LayoutProviderRegistry layoutProviderRegistry;
+
+    @Inject
+    private ArtifactMetadataService artifactMetadataService;
+
+    @Inject
+    private MavenMetadataManager mavenMetadataManager;
+
+    public void deleteMetadataInRepositoryParents(final String storageId,
```

Comment:
Parents? I think the name is a bit dubious... How about `deleteMetadataInGroupRepositoriesContainingPath`...?